src/test/scala/latis/service/hapi/DataServiceSuite.scala

```diff
@@ -0,0 +1,296 @@
+package latis.service.hapi
+
+import cats.data.NonEmptyList
+import cats.effect.IO
+import cats.syntax.all._
+import io.circe.Json
+import io.circe.syntax._
+import munit.CatsEffectSuite
+import org.http4s._
+import org.http4s.headers.`Content-Type`
+import org.http4s.implicits._
+import org.http4s.{Status => _}
+import scodec.codecs
+
+import latis.catalog.Catalog
+import latis.data.DomainData
+import latis.data.RangeData
+import latis.data.Sample
+import latis.data.SeqFunction
+import latis.dataset.MemoizedDataset
+import latis.metadata.Metadata
+import latis.model._
+import latis.service.hapi.HapiError._
+import latis.service.hapi.HapiInterpreter.noopInterpreter
+import latis.service.hapi.{Status => HStatus}
+import latis.time.Time
+import latis.util.Identifier.IdentifierStringContext
+
+class DataServiceSuite extends CatsEffectSuite {
+
+  /** Build a simple test DataService[IO] with a time -> int dataset using the Latis3Interpreter*/
+  private lazy val dataset = (for {
+    time <- Time.fromMetadata(Metadata("id"->"time", "type"->"string", "units"->"yyyy-MM-dd", "coverage"->"2000-01-01/2000-01-05"))
+    disp <- Scalar.fromMetadata(Metadata("id"->"displacement", "type"->"int", "units"->"meters"))
+    model <- Function.from(id"testfunc", time, disp)
+    data = new SeqFunction(Seq(
+      Sample(DomainData("2000-01-01"), RangeData(1)),
+      Sample(DomainData("2000-01-02"), RangeData(5)),
+      Sample(DomainData("2000-01-03"), RangeData(4)),
+      Sample(DomainData("2000-01-04"), RangeData(2)),
+      Sample(DomainData("2000-01-05"), RangeData(0)),
+    ))
+    dataset = new MemoizedDataset(Metadata("id"->"testdataset"), model, data)
+  } yield dataset).fold(err => fail(err.message), identity)
+  private lazy val latisInterp = new Latis3Interpreter(Catalog(dataset))
+  private lazy val dataService = new DataService[IO](latisInterp).service
+
+  /** Assert GET request to given URI returns a particular status. */
+  def assertStatus(uri: Uri, status: HStatus): IO[Unit] = {
```

Comment:
Should we include the implicit Location?
https://scalameta.org/munit/docs/tests.html#declare-tests-inside-a-helper-function

Reply:
Sure, might be useful